File: Python/array-partition-i.py
class Solution:
    def arrayPairSum1(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums.sort()
        res = 0
        for i in range(0, len(nums), 2):
            res += nums[i]
        return res

    def arrayPairSum2(self, nums):
        nums.sort()
        return sum([nums[i] for i in range(0, len(nums), 2)])

File: Swift/array-partition-i.swift
class Solution {
    func arrayPairSum1(_ nums: [Int]) -> Int {
        let nums = nums.sorted()
        var res = 0

        for (i, num) in nums.enumerated() {
            if i % 2 == 0 {
                res += num
            }
        }

        return res
    }

    func arrayPairSum2(_ nums: [Int]) -> Int {
        return nums.sorted().enumerated()
            .compactMap { $0 % 2 == 0 ? $1 : nil }
            .reduce(0) { $0 + $1 }
    }
}
